### DISCUSSION TITLE
Document platforms supported

### DISCUSSION THREAD

**Discussion by baentsch** [open-quantum-safe/liboqs#1583] [Ideas]
This discussion is to take up the suggestion in https://github.com/open-quantum-safe/liboqs/issues/1573#issuecomment-1765668329 and propose clearly documenting the hard- and software platforms this project supports.

I propose at least two levels:
- Fully supported and tested (OS+HW) platform combinations: There must be a CI environment fully checking these at latest during merge; ideally adherence to this goal is automatically verified; the CI environment may be either within this project or a (maintained) downstream integration;
- Best effort platform combinations: Confirmed as locally tested by committer; it should be possible to at least cross-build them in CI; best would be an actual CI test run -- even if only done weekly in a slow, emulated environment.

For each support level there also should be a documented procedure as to how platforms are getting added/removed from the list.

Proposals for alternative/further platform support types welcome. 

In a second step we should open a discussion as to which platforms should be added under which support level type. Finally, the fully supported platforms should be scrutinized as to whether they are sufficiently tested based on the criteria defined.

Without feedback in this discussion, I'll proceed to implementation of this proposal by way of creating a PR for a new file "PLATFORMS.md", two sections with the proposal above and initial lists of supported platforms.

**Comment by baentsch** [comment]
Silly to comment my own proposal, I know, but what about adding an intermediate support tier: Works in CI but is known to have "corner case" issues (resolved for the "really supported" platforms)? Windows is a clear candidate for that tier. Maybe call it "Home" or "361"? :) The one above would be "Professional" and the one below "Yolo". Other proposals welcome.

**Comment by thomwiggers** [comment]
For reference, https://doc.rust-lang.org/beta/rustc/platform-support.html might be a nice example of how the Rust project treats this.

**Comment by dstebila** [comment]
Thanks @thomwiggers, the Rust platform tiers are really helpful to see.

@baentsch do you feel your three levels (fully supported/some corner cases/best effort) map on to the Tier 1/2/3 categories in Rust?  If anything your third level seems to offer higher promises than Rust Tier 3...

**Reply by baentsch** [reply]
> your third level seems to offer higher promises than Rust Tier 3...

For Tier 3 I went with the Rust text in https://github.com/open-quantum-safe/liboqs/pull/1605 and effectively merging "my" two lower tiers into Tier 2.
